src/figuresClasses.ts

```diff
@@ -1,11 +1,79 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sizes: number[] = [a, b, c];
+    const maxSize: number | undefined = sizes
+      .sort((aSide: number, bSide: number) => {
+        return aSide - bSide;
+      })
+      .pop();
+    const lowerSidesSum = sizes.reduce((acc, side) => acc + side, 0);
+
+    if (a <= 0 || b <= 0 || c <= 0 || (maxSize && lowerSidesSum <= maxSize)) {
+      throw new Error('Triangle side cannot be <= 0');
+    }
+  }
+
+  getArea(): number {
```

Comment:
The 'getArea' method should not accept any parameters according to the checklist. All necessary values should be accessed as object fields, which you are already doing.